Modules/_asynciomodule.c

```diff
@@ -2213,7 +2233,8 @@ _asyncio_Task___init___impl(TaskObj *self, PyObject *coro, PyObject *loop,
         // optimization: defer task name formatting
         // store the task counter as PyLong in the name
         // for deferred formatting in get_name
-        name = PyLong_FromUnsignedLongLong(++state->task_name_counter);
+        FT_ATOMIC_ADD_UINT64(state->task_name_counter, 1);
+        name = PyLong_FromUnsignedLongLong(state->task_name_counter);
```

Comment:
There's still a data race here due to the non-atomic read of `state->task_name_counter`.

I think we should just use a `#ifdef Py_GIL_DISABLED` here, something like:

```
#ifdef Py_GIL_DISABLED
Py_ssize_t counter = _Py_atomic_add_uint64(&state->task_name_counter, 1) + 1;
#else 
Py_ssize_t counter = ++state->task_name_counter;
#endif
name = PyLong_FromUnsignedLongLong(counter);
```